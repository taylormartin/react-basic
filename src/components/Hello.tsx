import * as React from 'react';
import Link from './Link';

export interface HelloProps { compiler: string; framework: string; }

export default class Hello extends React.Component<HelloProps, any> {

  constructor(props: HelloProps) {
    super(props);
    this.state = {
      class: 'black',
      linkClass: 'black'
    };
  }

  private onMouseEnter = (): void => {
    this.setState({class: 'red'});
  }

  private onMouseLeave = (): void => {
    this.setState({class: 'black'});
  }

  private toggleLinkColor = (): void => {
    let color = this.state.linkClass === 'black' ? 'green' : 'black';
    this.setState({linkClass: color});
  }

  public render() {
    return (
      <div>
        <h1 onMouseLeave={this.onMouseLeave} onMouseEnter={this.onMouseEnter} className={this.state.class}>Hello from {this.props.compiler} and {this.props.framework}!</h1>
        <button onClick={this.toggleLinkColor}>Change Link Color</button>
        <Link colorClass={this.state.linkClass}/>
        <div>text</div>
        <div>more text</div>
        <ul>
          <li>item one</li>
          <li>item two</li>
        </ul>
      </div>
    );
  }
}
