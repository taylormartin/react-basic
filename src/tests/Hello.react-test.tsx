import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Hello from '../components/Hello';

test('h1 changes class when hovered', () => {

  const component = renderer.create(
    <Hello compiler="Typescript" framework="React" />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseEnter();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});