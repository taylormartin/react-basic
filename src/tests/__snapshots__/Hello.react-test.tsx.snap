// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h1 changes class when hovered 1`] = `
<div>
  <h1
    className="black"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Hello from 
    Typescript
     and 
    React
    !
  </h1>
</div>
`;

exports[`h1 changes class when hovered 2`] = `
<h1
  className="red"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Hello from 
  Typescript
   and 
  React
  !
</h1>
`;

exports[`h1 changes class when hovered 3`] = `
<h1
  className="black"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Hello from 
  Typescript
   and 
  React
  !
</h1>
`;
